### PR TITLE
Support a "labels" field in AxisInfo if present. 

### PR DESCRIPTION
**Pull request by cboulay**
ezmsg-lsl has a [SpaceAxis that I patch into AxisArray](https://github.com/labstreaminglayer/ezmsg-lsl/blob/62d1458bb335b82d7d7b5c7930d5770c4463ee70/src/ezmsg/lsl/units.py#L13-L23), and SpaceAxis has a `labels` field. This solves an issues that we discussed in #43 

Units that transform the AxisArray in a way that modifies the length of the dimension with the "labels" field, should probably also modify the labels, if present! This PR does that for Units that I'm aware of that modify the shape of the array along axes other than "time".

For `affine_transform`, it's difficult to know what the new labels should be so in most cases they are just dropped. However, if the `affine_transform` has rows or columns of zeros then we can get a good guess at thew new labels.

### CODE REVIEW

**Comment by griffinmilsap** [comment]
I didn't know about an lsl connector; this is really cool!  When you're ready, let's add it into the README as another extension :D

AxisArray is VERY inspired by [XArray](https://github.com/pydata/xarray) (my absolutely favorite data structure for offline analyses) that unfortunately has a little too much going on under the hood to be performant enough for low latency, high-message rate stream processing.  An earlier version of the ezmsg-sigproc lib actually used DataArrays for the messages and that was a lovely developer experience even if my CPUs pegged processing 8 channels of 250 Hz EEG data -- but I digress.

I feel like the general solution to this particular problem is adding a `coords` implementation into AxisArray that captures some core set of functionality that we find in XArray's DataArray `coords`.  I wrote the majority of the AxisArray implementation in an afternoon when I was pressured by an incoming deadline.  I spent about an hour thinking about how to get `coords` in without destroying performance or my guiding mantra that "AxisArray is simply a metadata wrapper around a numpy array".  I came to the conclusion at that time that a limited `coords` functionality in AxisArray would 1. be incredibly useful, 2. could be performant if we removed all the integrity checks/safety rails that XArray's implementation has.  I simply didn't have the bandwidth (at that time) to even do a first-pass implementation.

In summary, AxisArray needs some TLC for 
1. Better Typehinting (even simple fixes like sel/isel returning subclasses of AxisArray, not just AxisArrays)
2. Better dev experience/consistency (why is concatenate a `staticmethod`?) 
3. A lightweight `coords` implementation

__Coming back to today and this PR:__ These operations on coords/labels probably eventually belong in AxisArray helper functions/implementation rather than spread throughout the signal processing modules.  Collecting common operations on AxisArrays together in one place would make this whole thing a lot easier to maintain, but if this gets you where you're going _today_ I support a merge after review (feel free to merge it in yourself once my review posts)

File: extensions/ezmsg-sigproc/src/ezmsg/sigproc/affinetransform.py [hunk 1]
```diff
@@ -48,6 +51,43 @@ def affine_transform(
...
+                    elif np.all(b_used_inputs):
+                        # Strange scenario: New outputs are filled with empty data.
+                        in_ix = 0
+                        new_labels = []
+                        for out_ix in range(n_out):
+                            if b_filled_outputs[out_ix]:
+                                new_labels.append(in_labels[in_ix])
+                                in_ix += 1
+                            else:
+                                new_labels.append("")
```
Comment (griffinmilsap) [inline extensions/ezmsg-sigproc/src/ezmsg/sigproc/affinetransform.py:85]: Maybe append None here instead?  We could also assume `labels` will always be a `typing.List[str]` if it exists

Comment (griffinmilsap) [inline extensions/ezmsg-sigproc/src/ezmsg/sigproc/affinetransform.py:85]: Maybe append `None` here instead? If you keep `""`, we should be explicit that we assume labels will always be a `typing.List[str]` if it exists

**Review by griffinmilsap — COMMENTED** [review]
Looks alright to me :). Maybe consider adding your monkeypatch into AxisArray while you're at it.

**Review by griffinmilsap — APPROVED** [review]

File: extensions/ezmsg-sigproc/src/ezmsg/sigproc/affinetransform.py (hunk 1 above)
Comment (cboulay) [reply extensions/ezmsg-sigproc/src/ezmsg/sigproc/affinetransform.py:85]: Yes, it is always a `typing.List[str]`, not a `typing.List[typing.Optional[str]]`. The `""` is intended to indicate that we don't know what the label is. If anything, I was thinking about replacing the `""` with `"?"` or even `str(len(new_labels)+1)`. But I decided against these because integers might already have a different meaning, and I just know that "?" is going to interfere with somebody's localization or other parser down the line eventually.

**Comment by cboulay** [comment]
Moved to https://github.com/ezmsg-org/ezmsg-sigproc/pull/1
